src/main/webapp/js/script.js

```diff
@@ -18,27 +18,35 @@ async function getPlatforms() {
   }
 }
 
-// Fetch favorite SDKS
-async function getFavoriteSdks(userID) {
-  const response = await fetch(`v1/users/${userID}/sdks`);
-  const favorites = await response.json();
-  // Build the links for favorite sdks
-  const favoriteSDK = document.getElementById('favorites');
-  console.log(favoriteSDK);
-  for (i = 0; i < favorites.length; i++) {
-    favoriteSDK.appendChild(createListElement(favorites[i]));
-    favoriteSDK.appendChild(document.createElement("HR"));
-    console.log(favorites[i]);
-  }
-}
-
 // Create link for favorite sdk
 function createListElement(text) {
-  const sdkElement = document.createElement('a');
+  const sdkElement = document.createElement('div');
   sdkElement.innerText = text;
   sdkElement.setAttribute('class','user-favorite');
-  sdkElement.setAttribute('href','product_release.html');
   return sdkElement;
 }
 
 getPlatforms();
+
+firebase.auth().onAuthStateChanged(async function(user) {
+  if (user) {
+    const response = await fetch(`v1/users/${user.uid}/sdks`);
+    const favorites = await response.json();
+    console.log(favorites);
+    // Build the links for favorite sdks
+    const favoriteSDK = document.getElementById('favorites');
+    console.log(favoriteSDK);
+    for (i = 0; i < favorites['ANDROID'].length; i++) {
+      const sdkName = favorites['ANDROID'][i];
+      favoriteSDK.appendChild(createListElement(favorites['ANDROID'][i]));
+      favoriteSDK.appendChild(document.createElement("HR"));
+      favoriteSDK.addEventListener("click", () => {
+        const params = new URLSearchParams(`platform=android&sdkName=${sdkName}`);
+        const page = "release_history.html";
+        goToPage(page, params);
+      });
+
+      console.log(favorites[i]);
```

Comment:
This looks similar to the code in getFavoriteSDKs method. Can you avoid repeating the code and call it here?
Or delete any code that is unnecessary in the getFavoriteSDKs method.